library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -361,6 +361,11 @@ public Slice get(Slice key, long timeBucket, ReadSource readSource)
      */
     private BucketedValue getValueFromTimeBucketReader(Slice key, long timeBucket)
     {
+
+      if (managedStateContext.getTimeBucketAssigner() instanceof MovingBoundaryTimeBucketAssigner &&
```

Comment:
A comment is appropriate why we need this special case for a specific time bucket assigner. What happens if it is a different assigner. Please improve the javadoc if this method as well.